packages/block-library/src/search/index.php

```diff
@@ -360,13 +378,118 @@ function styles_for_block_core_search( $attributes ) {
+		}
+	}
+
+	$has_letter_spacing = ! empty( $attributes['style']['typography']['letterSpacing'] );
+	if ( $has_letter_spacing ) {
+		$letter_spacing_value = sprintf( 'letter-spacing: %s;', esc_attr( $attributes['style']['typography']['letterSpacing'] ) );
+		$button_styles[]      = $letter_spacing_value;
+		$input_styles[]       = $letter_spacing_value;
+		if ( $show_label ) {
+			$label_styles[] = $letter_spacing_value;
+		}
+	}
+
+	$has_font_weight = ! empty( $attributes['style']['typography']['fontWeight'] );
+	if ( $has_font_weight ) {
+		$font_weight_value = sprintf( 'font-weight: %s;', esc_attr( $attributes['style']['typography']['fontWeight'] ) );
+		$button_styles[]   = $font_weight_value;
+		$input_styles[]    = $font_weight_value;
+		if ( $show_label ) {
+			$label_styles[] = $font_weight_value;
+		}
+	}
+
+	$has_font_style = ! empty( $attributes['style']['typography']['fontStyle'] );
+	if ( $has_font_style ) {
+		$font_style_value = sprintf( 'font-style: %s;', esc_attr( $attributes['style']['typography']['fontStyle'] ) );
+		$button_styles[]  = $font_style_value;
+		$input_styles[]   = $font_style_value;
+		if ( $show_label ) {
+			$label_styles[] = $font_style_value;
+		}
+	}
+
+	$has_line_height = ! empty( $attributes['style']['typography']['lineHeight'] );
+	if ( $has_line_height ) {
+		$line_height_value = sprintf( 'line-height: %s;', esc_attr( $attributes['style']['typography']['lineHeight'] ) );
+		$button_styles[]   = $line_height_value;
+		$input_styles[]    = $line_height_value;
+		if ( $show_label ) {
+			$label_styles[] = $line_height_value;
+		}
+	}
+
+	$has_text_transform = ! empty( $attributes['style']['typography']['textTransform'] );
+	if ( $has_text_transform ) {
+		$text_transform_value = sprintf( 'text-transform: %s;', esc_attr( $attributes['style']['typography']['textTransform'] ) );
+		$button_styles[]      = $text_transform_value;
+		$input_styles[]       = $text_transform_value;
+		if ( $show_label ) {
+			$label_styles[] = $text_transform_value;
+		}
+	}
+
+	$has_text_decoration = ! empty( $attributes['style']['typography']['textDecoration'] );
+	if ( $has_text_decoration ) {
+		$text_decoration_value = sprintf( 'text-decoration: %s;', esc_attr( $attributes['style']['typography']['textDecoration'] ) );
+		$button_styles[]       = $text_decoration_value;
+		// Input opts out of text decoration.
+		if ( $show_label ) {
+			$label_styles[] = $text_decoration_value;
+		}
+	}
+
 	return array(
 		'input'   => ! empty( $input_styles ) ? sprintf( ' style="%s"', safecss_filter_attr( implode( ' ', $input_styles ) ) ) : '',
 		'button'  => ! empty( $button_styles ) ? sprintf( ' style="%s"', safecss_filter_attr( implode( ' ', $button_styles ) ) ) : '',
 		'wrapper' => ! empty( $wrapper_styles ) ? sprintf( ' style="%s"', safecss_filter_attr( implode( ' ', $wrapper_styles ) ) ) : '',
+		'label'   => ! empty( $label_styles ) ? sprintf( ' style="%s"', safecss_filter_attr( implode( ' ', $label_styles ) ) ) : '',
```

Comment:
```suggestion
		'label'   => ! empty( $label_styles ) ? sprintf( ' style="%s"', esc_attr( safecss_filter_attr( implode( ' ', $label_styles ) ) ) ) : '',
```

There's an open PR that should merge soon that already addresses the other styles. It won't address the new `label` styles though.